Community-Supported/flights-data-incremental-refresh/flights-data-incremental-refresh.py

```diff
@@ -0,0 +1,118 @@
+    """
+    # Create a tableuserverclient object to interact with Tableau Server.
+    server = TSC.Server(server_url, use_server_version=True)
+    # Sign into Tableau Server with the above authentication information.
+    with server.auth.sign_in(tableau_auth):
+        # Get project_id from project_name.
+        matching_projects = server.projects.filter(name=project_name)
+        project_id = next((project.id for project in matching_projects if project.name == project_name), None)
+        if project_id is None:
+            print(f"Publish failed. The specified project '{project_name}' does not exist.")
+            exit()
+
+        # Get the datasource from Server (if it exists).
+        matching_datasources = server.datasources.filter(name=datasource_name_on_server)
+        datasource = next((ds for ds in matching_datasources), None)
+
+        if datasource is None:
+            # If the datasource does not exist on server, publish the datasource. 
+            publish_mode = TSC.Server.PublishMode.CreateNew
+            datasource = TSC.DatasourceItem(project_id)
+            # Set the name of the datasource such that it can be easily identified.
+            datasource.name = datasource_name_on_server
+            datasource = server.datasources.publish(datasource, database_path, publish_mode)
+            print(f"New datasource published: (id : {datasource.id}, name: {datasource.name}).")
+        else:
+            # If the datasource already exists on Tableau Server, use the Hyper Update REST API 
+            # to send the delta to Tableau Server and insert the data into the respective table
+            # in the datasource. 
+
+            # Create a new random request id. 
+            request_id = str(uuid.uuid4())
+
+            # Create one action that inserts from the new table into the existing table.
+            actions = [
+                {
+                    "action": "insert",
+                    "source-schema": "public",
+                    "source-table": "flights",
+                    "target-schema": "public",
+                    "target-table": "flights",
+                }
+            ]
+
+            # Start the update job on Server.
+            job = server.datasources.update_hyper_data(datasource.id, request_id=request_id, actions=actions, payload=database_path)
+            print(f"Update job posted (ID: {job.id}). Waiting for the job to complete...")
+
+            # Wait for the job to finish.
+            job = server.jobs.wait_for_job(job)
+            print("Job finished successfully")  
+       
+        
+if __name__ == '__main__':
+    # First create the hyper database with flights data.
+    database_path = "flights.hyper"
+    create_hyper_database_with_flights_data(database_path)
+
+    # Then publish the data to server.
+    server_url = ''
+    site_name = ''
+    project_name = ''
+    token_name = ''
+    token_value = ''
+    datasource_name_on_server = 'flights_data_set'
```

Comment:
all other samples use the argument parser and provide a command line interface. I think we should do the same here, too